src/lib/tuning/index/base_index_evaluator.cpp

```diff
@@ -0,0 +1,222 @@
+#include "base_index_evaluator.hpp"
+
+#include <algorithm>
+#include <list>
+#include <memory>
+#include <string>
+
+#include "logical_query_plan/abstract_lqp_node.hpp"
+#include "logical_query_plan/predicate_node.hpp"
+#include "logical_query_plan/stored_table_node.hpp"
+#include "operators/get_table.hpp"
+#include "operators/table_scan.hpp"
+#include "operators/validate.hpp"
+#include "optimizer/column_statistics.hpp"
+#include "optimizer/table_statistics.hpp"
+#include "sql/gdfs_cache.hpp"
+#include "sql/sql_query_cache.hpp"
+#include "storage/index/base_index.hpp"
+#include "storage/storage_manager.hpp"
+#include "types.hpp"
+#include "utils/logging.hpp"
+
+namespace opossum {
+
+BaseIndexEvaluator::BaseIndexEvaluator() {}
+
+void BaseIndexEvaluator::evaluate(std::vector<std::shared_ptr<TuningChoice>>& choices) {
+  // Allow concrete implementation to initialize
+  _setup();
+
+  // Scan query cache for indexable table column accesses
+  _inspect_query_cache();
+
+  // Aggregate column accesses to set of new columns to index
+  _aggregate_access_records();
+
+  // Fill index_evaluations vector
+  _choices.clear();
+  _add_existing_indexes();
+  _add_new_indexes();
+
+  // Evaluate
+  for (auto& index_choice : _choices) {
+    if (index_choice.exists) {
```

Comment:
Naming: Does the choice exist or the index?

Reply:
renamed to `index_exists`